step-70: warning: possibly dangling reference to a temporary [-Wdangling-reference]

I built the library with the component examples recently with `gcc 13.1.1` and I got the following warning for step-70:
```
Building CXX object examples/CMakeFiles/example_step_70_debug.dir/step-70/step-70.cc.o
dealii/examples/step-70/step-70.cc: In instantiation of ‘void Step70::StokesImmersedProblem<dim, spacedim>::assemble_nitsche_restriction() [with int dim = 2; int spacedim = 3]’:
dealii/examples/step-70/step-70.cc:1869:9:   required from ‘void Step70::StokesImmersedProblem<dim, spacedim>::run() [with int dim = 2; int spacedim = 3]’
dealii/examples/step-70/step-70.cc:1938:22:   required from here
dealii/examples/step-70/step-70.cc:1484:25: warning: possibly dangling reference to a temporary [-Wdangling-reference]
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                         ^~~
dealii/examples/step-70/step-70.cc:1484:52: note: the temporary was destroyed at the end of the full expression ‘(& p)->dealii::Particles::ParticleAccessor<3, 3>::get_properties().dealii::ArrayView<const double>::operator[](0)’
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                                  ~~~~~~~~~~~~~~~~~~^
dealii/examples/step-70/step-70.cc: In instantiation of ‘void Step70::StokesImmersedProblem<dim, spacedim>::assemble_nitsche_restriction() [with int dim = 3; int spacedim = 3]’:
dealii/examples/step-70/step-70.cc:1869:9:   required from ‘void Step70::StokesImmersedProblem<dim, spacedim>::run() [with int dim = 3; int spacedim = 3]’
dealii/examples/step-70/step-70.cc:1946:22:   required from here
dealii/examples/step-70/step-70.cc:1484:25: warning: possibly dangling reference to a temporary [-Wdangling-reference]
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                         ^~~
dealii/examples/step-70/step-70.cc:1484:52: note: the temporary was destroyed at the end of the full expression ‘(& p)->dealii::Particles::ParticleAccessor<3, 3>::get_properties().dealii::ArrayView<const double>::operator[](0)’
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                                  ~~~~~~~~~~~~~~~~~~^
dealii/examples/step-70/step-70.cc: In instantiation of ‘void Step70::StokesImmersedProblem<dim, spacedim>::assemble_nitsche_restriction() [with int dim = 2; int spacedim = 2]’:
dealii/examples/step-70/step-70.cc:1869:9:   required from ‘void Step70::StokesImmersedProblem<dim, spacedim>::run() [with int dim = 2; int spacedim = 2]’
dealii/examples/step-70/step-70.cc:1954:22:   required from here
dealii/examples/step-70/step-70.cc:1484:25: warning: possibly dangling reference to a temporary [-Wdangling-reference]
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                         ^~~
dealii/examples/step-70/step-70.cc:1484:52: note: the temporary was destroyed at the end of the full expression ‘(& p)->dealii::Particles::ParticleAccessor<2, 2>::get_properties().dealii::ArrayView<const double>::operator[](0)’
 1484 |             const auto &JxW    = p.get_properties()[0];
      |                                  ~~~~~~~~~~~~~~~~~~^
```

--- a/examples/step-70/step-70.cc
+++ b/examples/step-70/step-70.cc
@@ -1479,9 +1479,9 @@ namespace Step70
         Assert(pic.begin() == particle, ExcInternalError());
         for (const auto &p : pic)
           {
-            const auto &ref_q  = p.get_reference_location();
-            const auto &real_q = p.get_location();
-            const auto &JxW    = p.get_properties()[0];
+            const Point<spacedim> ref_q  = p.get_reference_location();
+            const Point<spacedim> real_q = p.get_location();
+            const double          JxW    = p.get_properties()[0];
 
             for (unsigned int i = 0; i < fluid_fe->n_dofs_per_cell(); ++i)
               {

